ditto/readers/sincal/read_photovoltaics.py

```diff
@@ -0,0 +1,411 @@
+from __future__ import absolute_import, division, print_function
+from builtins import super, range, zip, round, map
+import logging
+import math
+import sys
+import os
+import json
+import cmath
+import sqlite3
+from sqlite3 import Error
+import math
+import numpy as np
+import threading
+
+logger = logging.getLogger(__name__)
+
+from ditto.readers.abstract_reader import AbstractReader
+from ditto.store import Store
+from ditto.models.node import Node
+from ditto.models.line import Line
+from ditto.models.load import Load
+from ditto.models.phase_load import PhaseLoad
+from ditto.models.position import Position
+from ditto.models.power_source import PowerSource
+from ditto.models.powertransformer import PowerTransformer
+from ditto.models.winding import Winding
+from ditto.models.phase_winding import PhaseWinding
+from ditto.models.regulator import Regulator
+from ditto.models.wire import Wire
+from ditto.models.capacitor import Capacitor
+from ditto.models.phase_capacitor import PhaseCapacitor
+from ditto.models.reactor import Reactor
+from ditto.models.phase_reactor import PhaseReactor
+from ditto.models.photovoltaic import Photovoltaic
+from ditto.readers.sincal.exception_logger import log_exceptions
+
```

Comment:
There are two methods. One parses a single element. The other parses numerous instances of that element, calling the single element method iteratively.